src/components/changePlanner/CPDetailView.tsx

```diff
@@ -164,22 +173,45 @@ class CPDetailView extends React.Component<
         this.addErrorToast(err.response.data.failure_message);
       });
   }
-
+  handleExecuteCancel() {
+    this.setState({
+      isAlertOpen: false
+    });
+  }
   handleExecute() {
     axios
       .post(
-        API_ROOT +
-        "api/change-plans/" +
-        this.state.changePlan.id +
-        "/execute",
+        API_ROOT + "api/change-plans/" + this.state.changePlan.id + "/execute",
         {},
         getHeaders(this.props.token)
       )
       .then(res => {
         this.addSuccessToast(res.data.success_message);
+        this.setState({
+          isAlertOpen: false
+        });
+
+        this.loading = true;
+        getChangePlanDetail(this.props.token, this.route_id)
+          .then(res => {
+            this.loading = false;
+            this.dataLoaded = true;
+            this.props.setChangePlan(res.data.change_plan);
+            this.setState({
+              changePlan: res.data.change_plan,
```

Comment:
if change plan has been executed, we should not enter the change plan automatically when going into the detail view, bc that introduces a hacky way to get into the change plan

Reply:
^^ i think this is the most important change, if you fix it ill approve since the rest is low priority

Reply:
ooooooooooooooooo good CATCH

Reply:
expected behavior now is that if you click onto a change plan that has been executed ,you are automatically not on a change plan.